account-integrations/sdk/src/accounts/SafeECDSAAccontWrapper.ts

```diff
@@ -0,0 +1,249 @@
+      data.address,
+      data.privateKey,
+      data.ownerAddress,
+      data.recoveryAddress,
+      provider,
+    );
+  }
+
+  toData(): SafeECDSAAccountData {
+    return {
+      type: 'SafeECDSAAccount',
+      address: this.address,
+      privateKey: this.privateKey,
+      ownerAddress: this.ownerAddress,
+      recoveryAddress: this.recoveryAddress,
+    };
+  }
+
+  static async createRandom(
+    // TODO: clean up parameters needed for getContracts
+    provider: ethers.BrowserProvider,
+    admin: ethers.Signer
+  ): Promise<SafeECDSAAccountWrapper> {
+    const contracts = await getContracts(provider, admin);
+
+    const wallet = ethers.Wallet.createRandom();
+
+    const createArgs = [
+      contracts.safe,
+      contracts.entryPoint,
+      wallet,
+      0,
+    ] satisfies Parameters<SafeECDSAFactory['create']>;
+
+    const address = await contracts.safeECDSAFactory.create.staticCall(
+      ...createArgs,
+    );
+
+    await receiptOf(
+      contracts.safeECDSAFactory.connect(admin).create(...createArgs),
+    );
+
+    return new SafeECDSAAccountWrapper(
+      address,
+      wallet.privateKey,
+      wallet.address,
+      undefined,
+      provider,
+    );
+  }
+
+  getContract(): SafeECDSAPlugin {
+    return SafeECDSAPlugin__factory.connect(
+      this.address,
+      this.provider
+    );
+  }
+
+  // eslint-disable-next-line class-methods-use-this, @typescript-eslint/require-await
+  async makeInitCode(): Promise<string> {
+    throw new Error(
+      'SafeECDSAAccount does not use initCode (it must be created before use)',
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await, class-methods-use-this
+  async encodeActions(actions: EthereumRpc.Action[]): Promise<string> {
+    if (actions.length !== 1) {
+      throw new Error('SafeECDSAAccount only supports single actions');
+    }
+
+    const action = actions[0];
+
+    return SafeECDSAPlugin__factory.createInterface().encodeFunctionData(
+      'execTransaction',
+      [action.to, action.value ?? 0n, action.data ?? '0x'],
+    );
+  }
+
+  // eslint-disable-next-line class-methods-use-this, @typescript-eslint/require-await
+  async estimateVerificationGas(
+    _userOp: EthereumRpc.UserOperation,
+  ): Promise<bigint> {
+    // TODO: estimateGas on validateUserOp?
+    return baseVerificationGas;
+  }
+
+  async getNonce(): Promise<bigint> {
+    const contracts = await getContracts(this.provider);
+
+    // TODO: Why does this give a different result to
+    // this.getContract().getNonce()?
+    // (And why does that alternative give the wrong answer?)
+    return await contracts.entryPoint.getNonce(this.address, 0);
+  }
+
+  async sign(
+    _userOp: EthereumRpc.UserOperation,
+    userOpHash: string,
+  ): Promise<string> {
+    const ownerWallet = new ethers.Wallet(this.privateKey);
+
+    return await ownerWallet.signMessage(ethers.getBytes(userOpHash));
+  }
+
+  // TODO: update this with the new recovery module
```

Comment:
Didn't get a chance to update this logic with the recovery plugin changes.  We will need to do that